projects/schematics/src/shared/utils/file-utils_spec.ts

```diff
@@ -176,20 +203,313 @@ describe('File utils', () => {
+        )
+      ).toEqual(false);
+    });
+    it('should return false if the constructor condition is not satisfied', () => {
+      const content = `
+      import { UserAddressService } from '@spartacus/core';
+      import { Store } from '@ngrx/store';
+      export class InheritingService extends UserAddressService {}
+      `;
+      const source = ts.createSourceFile(
+        'xxx.ts',
+        content,
+        ts.ScriptTarget.Latest,
+        true
+      );
+      const parameterClassTypes: ClassType[] = [
+        { className: STORE, importPath: NGRX_STORE },
+      ];
+
+      expect(
+        isCandidateForConstructorDeprecation(
+          source,
+          USER_ADDRESS_SERVICE,
+          parameterClassTypes
+        )
+      ).toEqual(false);
+    });
+    it('should return false if the parameter lengths condition is not satisfied', () => {
+      const content = `
+      import { Store } from '@ngrx/store';
+      import {
+        AuthService,
+        StateWithProcess,
+        StateWithUser,
+        UserAddressService
+      } from '@spartacus/core';
+      export class InheritingService extends UserAddressService {
+        constructor(
+          authService: AuthService,
+          store: Store<StateWithUser | StateWithProcess<void>>
+        ) {
+          super(authService, store);
+        }
+      }
+      `;
+      const source = ts.createSourceFile(
+        'xxx.ts',
+        content,
+        ts.ScriptTarget.Latest,
+        true
+      );
+      const parameterClassTypes: ClassType[] = [
+        { className: STORE, importPath: NGRX_STORE },
+      ];
+
+      expect(
+        isCandidateForConstructorDeprecation(
+          source,
+          USER_ADDRESS_SERVICE,
+          parameterClassTypes
+        )
+      ).toEqual(false);
+    });
+    it('should return false if the parameter order is not satisfied', () => {
+      const content = `
+      import { Store } from '@ngrx/store';
+      import {
+        AuthService,
+        StateWithProcess,
+        StateWithUser,
+        UserAddressService
+      } from '@spartacus/core';
+      export class InheritingService extends UserAddressService {
+        constructor(
+          authService: AuthService,
+          store: Store<StateWithUser | StateWithProcess<void>>
+        ) {
+          super(authService, store);
+        }
+      }
+      `;
+      const source = ts.createSourceFile(
+        'xxx.ts',
+        content,
+        ts.ScriptTarget.Latest,
+        true
+      );
+      const parameterClassTypes: ClassType[] = [
+        { className: STORE, importPath: NGRX_STORE },
+        { className: USER_ADDRESS_SERVICE, importPath: SPARTACUS_CORE },
+      ];
+
+      expect(
+        isCandidateForConstructorDeprecation(
+          source,
+          USER_ADDRESS_SERVICE,
+          parameterClassTypes
+        )
+      ).toEqual(false);
+    });
+    it('should return false if the super() call does NOT exist', () => {
+      const content = `
```

Comment:
Can we go with consts before tests (as in `update-cms-component-state_spec.ts`)? Just for more consistency